server/ai_live_video.go

```diff
@@ -228,6 +238,45 @@ func startTrickleSubscribe(ctx context.Context, url *url.URL, params aiRequestPa
 			time.Sleep(5 * time.Second)
 		}
 	}()
+
+	// MediaMTX Output ffmpeg process
+	go func() {
+		defer func() {
+			rMediaMTX.Close()
+			if rec := recover(); rec != nil {
+				// panicked, so shut down the stream and handle it
+				err, ok := rec.(error)
+				if !ok {
+					err = errors.New("unknown error")
+				}
+				clog.Errorf(ctx, "LPMS panic err=%v", err)
+				params.liveParams.stopPipeline(fmt.Errorf("LPMS panic %w", err))
+			}
+		}()
+		for {
+			clog.V(6).Infof(ctx, "Starting MediaMTX output rtmp")
+			if !params.inputStreamExists() {
+				clog.Errorf(ctx, "Stopping MediaMTX output rtmp stream, input stream does not exist.")
+				break
+			}
+
+			cmd := exec.Command("ffmpeg",
+				"-i", "pipe:0",
```

Comment:
If the thread that pushes segments into the pipe:0 crashes, does this ffmpeg process die? what does recovery look like then?

Reply:
This was from the previous logic, and yep that's right, if the thread pushing segments stops then stopPipeline would have been called [here](https://github.com/livepeer/go-livepeer/blob/57dc10703f9d83a72ca59beb61d803c08aad2b4b/server/ai_live_video.go#L169) and so everything shuts down then the input stream can reconnect and try again.